lib/recommendation.js

```diff
@@ -0,0 +1,61 @@
+var XMLHttpRequest = Cc['@mozilla.org/xmlextras/xmlhttprequest;1'].createInstance();
+
+
+class RecommendationServer {
+  /*
+  Class providing a connection to a recommendation server and a method for
+  querying it. This class should be initialized once and exported as a
+  singleton to modules that will use it.
+
+  Usage:
+
+  let recommendation = new RecommendationServer('http://universal-search.dev');
+
+  */
+  constructor(server) {
+    this.server = server;
+  }
+
+  _url(query) {
+    return `${this.server}/?q=${encodeURIComponent(query)}`;
+  }
+
+  make(query) {
```

Comment:
You don't _search_ recommendations, you _make_ them. But that's just my twisted logic; happy to change :)